File: Tests/Tests.Antidote.FuzzySort.fs
module Tests.Fuse

open Mocha
open Node
open Fable.Core
open Fable.Core.Testing
open Fable.Core.JsInterop
open Antidote.FuzzySort


let haystack = ResizeArray [|
    "puzzle"
    "Super Awesome Thing (now with stuff!)"
    "FileName.js"
    "/feeding/the/catPic.jpg"
|]

let needle = "feed cat"

describe """uFuzzy.search""" (fun () ->

    it "works with strings" (fun () ->

        let result = fuzzysort.go(needle, haystack, null)
        Assert.AreEqual(result.total , 1)
    )
)

File: Tests/fableBuild/Tests.Antidote.FuzzySort.js
import fuzzysort from "fuzzysort";
import { some } from "./fable_modules/fable-library.3.7.8/Option.js";
import { assertEqual } from "./fable_modules/fable-library.3.7.8/Util.js";

export const haystack = ["puzzle", "Super Awesome Thing (now with stuff!)", "FileName.js", "/feeding/the/catPic.jpg"];

export const needle = "feed cat";

describe("uFuzzy.search", () => {
    it("works with strings", () => {
        const result = fuzzysort.go(needle, haystack, some(null));
        assertEqual(result.total, 1);
    });
});

